Add support for `:glob:` option to the `feed` directive

bosonogi opened the pull request:
It mimics the way `:glob:` works for the `toctree` directive.

bosonogi committed:
Add support for `:glob:` option to the `feed` directive

It mimics the way `:glob:` works for the `toctree` directive.

doismellburning commented:
Thank you! But I think this might have a bug around how paths are constructed.

The following in `source/notebook/index.rst` (adapted from a very very similar `toctree` directive where `*` just worked):

```
.. feed::
    :title: Notebook
    :rss: notebook.rss
    :glob:

    *
```

where e.g. I'm looking to include `source/notebook/2025-04-28-whats-a-backlog-anyway.rst`, gives me Sphinx build errors of the form:

```
/Users/me/git/bliki/source/notebook/index.rst:20: WARNING: feed contains a reference to nonexisting document 'notebook/notebook/2025-04-28-whats-a-backlog-anyway' [docutils]
```

(Note the double `notebook/notebook`)

I can't yet work out if you're accidentally adding something twice or not dropping something though...

bosonogi committed:
Do not modify names globed from `all_docnames`

If such names go through an additional `docname_join` call they might not be valid anymore.

bosonogi commented:
Somehow this did not affect my file layout, but I see the issue now. Documents are globed from a list of previously collected documents, and so those names should not be modified. Only the names that are _not_ glob patterns need to be joined with `env.docname`.

doismellburning commented:
@bosonogi That does the job nicely, thanks!

You might like to also grab https://github.com/doismellburning/sphinxcontrib-newsfeed/commit/840e981ed86bf6216d04bb5b2f63fa24e2eba9d8 which adds support for `:reversed:`!
